common/src/main/java/io/seata/common/loader/EnhancedServiceLoader.java

```diff
@@ -352,8 +352,10 @@ private static void loadFile(Class<?> service, String dir, ClassLoader classLoad
             constructor.setAccessible(true);
             s = service.cast(constructor.newInstance(args));
         } else {
-            // default Constructor
-            s = service.cast(implClazz.newInstance());
+            // default Constructor (might be private, use ctor.newInstance() instead of class.newInstance())
+            Constructor<S> constructor = implClazz.getDeclaredConstructor();
+            constructor.setAccessible(true);
+            s = service.cast(constructor.newInstance());
```

Comment:
Should a class with private constructor be loaded with SPI ?

Reply:
It is true that should not. The scenario I just met is that my service provider class is a singleton class (with a private constructor and a getInstance method). I should not use singleton mode in service provider class. Moreover, I  think this line of code 
`constructor.setAccessible(true);`
should not break the encapsulation of a class either when an argument constructor is called.

Reply:
Yes.I agree with you.